tests/test_precision.py

```diff
@@ -146,6 +148,30 @@ def test_snippets():
         cmp_regimes,
     ) = stumpy.snippets(T, m, k, s=s, mpdist_T_subseq_isconstant=isconstant_custom_func)
 
+    if (
+        not np.allclose(ref_snippets, cmp_snippets) and not numba.config.DISABLE_JIT
+    ):  # pragma: no cover
+        # Revise fastmath flags by removing reassoc (to improve precision),
+        # recompile njit functions, and re-compute snippets.
+        core._calculate_squared_distance.targetoptions["fastmath"] = {
```

Comment:
Just thinking out loud here. Seeing `core._calculate_squared_distance.targetoptions["fastmath"] = ` in the code feels like a hack and, if we want to do the same thing for other STUMPY njitted functions, we'll have look in this test in order to remember "how did we do it before?". I would even be okay if we had a `fastmath.py` module where we:

```
import fastmath

fastmath.set(module, func, flags)

# or

fastmath.reset(func, flags)  # Returns the function to STUMPY default fastmath flags
```

To me, `core._calculate_squared_distance.targetoptions["fastmath"] = ` is not so clear when we need to maintain/read this in the future. So, I'm trying to make it easier on our future selves. So, I can probably agree that we shouldn't change `config.STUMPY_FASTMATH_FLAGS` but perhaps we should do:

```
fastmath.set(core, _calculate_squared_distance, config.STUMPY_FASTMATH_FLAGS)
```

How does that sound? I want to hear your thoughts.

Reply:
> if we want to do the same thing for other STUMPY njitted functions, we'll have look in this test in order to remember "how did we do it before?"

In fact I forgot to add the `<func>.targetoptions['fasthmath']=...` part a few times when I am doing some experiments locally. I think this is another evidence that says it is better to wrap a function around it.

> fastmath.reset(func, flags)

To reset the flags to default, I think it is better to remove the param `flags` to reduce human error. Instead, we should go with what you suggested previously, i.e. finding a way to get the default flag. So, I took a closer look at the function `cache.get_njit_funcs`. If I add `print(ast.dump(decorator))` in the following if-block

https://github.com/TDAmeritrade/stumpy/blob/44168732e0c073cefdc208d1352bb4ae183a0eb5/stumpy/cache.py#L49-L50

I can see:

```
decorator dumped: 
 Call(func=Name(id='njit', ctx=Load()), args=[], keywords=[keyword(arg='fastmath', value=Attribute(value=Name(id='config', ctx=Load()), attr='STUMPY_FASTMATH_TRUE', ctx=Load()))])
```

So, we can add something like this inside the if-block above:

```
fastmath_default = None
for item in decorator.keywords:
     if item.arg == "fastmath":
         fastmath_default = item.value.attr
         break
```

But, I think it might not be a good idea to to add `fastmath_default` as the third value in the tuple `(module_name, func_name)` because the purpose of the function `get_njit_funcs` is to just find the njit functions. Maybe we should have a separate function inside the proposed module that says `get_default`. Its code will mostly be the same as the `cache.get_njit_funcs`.

Reply:
@seanlaw 
I just noticed the `# or` part in the code provided in your previous comment.  So, I can work on `fastmath.set(module, func, flags=None)`. If `flags` is provided, we can just set it to that. If not, `flags` will be set to the default fastmath, which can be obtained by the approach proposed in my previous comment.  

How do you feel about this?

Reply:
Okay... I did an initial implementation to see how this may look like. I pushed the commits so that you can see the `fastmath.py` too. I added the function `get_default_fastmath(module_name)` which returns a dictionary where a key is the name of a function and its corresponding value is its default fastmath flag. The function `set_flag` will use this default fastmath to set the fastmath flag if it is not provided by the user.